src/software/geom/voronoi_util.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <boost/polygon/voronoi.hpp>
+
+#include "software/geom/circle.h"
+#include "software/geom/point.h"
+#include "software/geom/rectangle.h"
+
+using boost::polygon::voronoi_builder;
+using boost::polygon::voronoi_diagram;
+using boost::polygon::voronoi_vertex;
+
+struct VoronoiDiagram
+{
+    explicit VoronoiDiagram(std::vector<Point> points)
+    {
+        construct_voronoi(points.begin(), points.end(), &vd);
+        this->points = points;
+    }
+
+   private:
+    voronoi_diagram<double> vd;
+    std::vector<Point> points;
+
+   public:
+    const voronoi_diagram<double> &getVd() const
+    {
+        return vd;
+    }
+
+    const std::vector<Point> &getPoints() const
```

Comment:
could we change this to `getPointsUsedForGeneration` just to be super explict?

Reply:
done